javatrix/Matrix.java

```diff
@@ -688,6 +690,25 @@ public Matrix uminus()
     // helper methods
     //////////////////////////////////////////////////////
     /**
+     * This is just an array copy convenience method.
+     * @param array       The array to be copied.
+     * @return            A copy of the array
+     **/
+    private static double[][] copyArray(double[][] array)
+    {
+        double[][] copy = new double[array.length][array[0].length];

```

Comment:
Should we handle the case in which we are asked to copy 0-length arrays? I suppose it would be silly...